helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl

```diff
@@ -150,7 +152,12 @@ service:
         - memory_limiter
         - k8sattributes
         - batch
+        {{- if or (hasPrefix "gke" (include "splunk-otel-collector.distribution" .)) (hasPrefix "eks" (include "splunk-otel-collector.distribution" .)) }}
```

Comment:
I was thinking it might be good to create a central boolean variable called `isClusterNameOptional` (or something like that) that's simply set to be whether the distribution started with `eks` or `gke`. This would be a lot cleaner in lots re-used places, and would allow easier extension in the future.

This could be done by adding an undocumented variable in `values.schema.json` and set it to `true` when the distribution doesn't require the `clusterName` setting. I thought this might open us up to some bad behavior though if a user found the option and set it to `true`.

Would another place work to add some kind of internal variable that could be referenced in templates? Or some maybe the `values.schema.json` option is acceptable? I'm open to suggestions, I'd really prefer to reference a single bool rather than many duplicate if-checks.

Reply:
sounds reasonable to move this in a helper but not sure how `values.schema.json` can help. We usualy keep all the templating helpers in _common.tpl

Reply:
My Notes:
- Some Kubernetes environments will present the whole values.schema.json file as the information on how to use the chart.
- I think defining a validation helper is acceptable when the validation logic is complicated and/or can be reused in multiple places. We do something similar with the operator in these locations.
  - https://github.com/signalfx/splunk-otel-collector-chart/blob/79312ca68dc3dfda114ef45c4bc10a1a22bfdd99/helm-charts/splunk-otel-collector/templates/operator/instrumentation.yaml#L2
  - https://github.com/signalfx/splunk-otel-collector-chart/blob/79312ca68dc3dfda114ef45c4bc10a1a22bfdd99/helm-charts/splunk-otel-collector/templates/operator/_helpers.tpl#L6

Reply:
I've added a bool definition to the [helpers template file.](https://github.com/signalfx/splunk-otel-collector-chart/blob/main/helm-charts/splunk-otel-collector/templates/_helpers.tpl)